frontend/src/components/RequestSourceInputs.vue

```diff
@@ -226,6 +245,60 @@ const inputValues: WritableComputedRef<RequestSource> = computed({
   },
 });
 
+const computedRules = computed(() => {
+  const sourceChain = {
+    required,
+    $autoDirty: true,
+  };
+  const token = {
+    required,
+    $autoDirty: true,
+  };
+
+  if (selectedToken.value) {
+    const selectedTokenDecimals = selectedToken.value.value.decimals;
+    const min = UInt256.parse('0', selectedTokenDecimals);
+
+    const amount = {
+      $autoDirty: true,
+      required: helpers.withMessage('Amount is required', required),
+      minValue: helpers.withMessage('Invalid amount', minValueUInt256(min, selectedTokenDecimals)),
+      isMatchingDecimals: helpers.withMessage(
+        'Decimals out of boundary',
+        isMatchingDecimals(selectedTokenDecimals),
```

Comment:
Just a comment regarding the text messages here which, I assume, will be shown to the user.

As a user, if I see 
```
Invalid amount
Decimals out of boundary
```
it is not immediately (or at all) obvious where exactly the error is or what a valid amount looks like.
Contrast this to e.g.
```
Amount cannot be negative
Amount must be at least $x
The number of decimals must be at most $x.
```
The latter example is much more specific and hints me, a user, what I need to do to make correct the error.

Reply:
Thanks for the great input Ivan!
Will adjust them accordingly.